### PR TITLE
feat(schedule): revamp schedule display in grid

### PR DESCRIPTION
**Pull request by flynnhou**
(Thanks for sending a pull request! Please fill in the following content to let us know better about this change.)

## Types of changes
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
Based on the [wireframe](https://docs.google.com/document/d/1sxfUqtaPaaJjnVXBvcazdEXa2oQrUPnZ9VwBoo_jkN4) provided by the design team:
- Display the schedule in grid system to provide users a glimpse of overall events including break time etc.
- Tweak events data in `views.py` to help frontend display the grid layout easier
- Take the liberty temporarily to design the style of the events and the schedule

> There are still some details to better realize the ideal of the wireframe

## Steps to Test This Pull Request
1. Go to the `/admin` to make sure there are events created just like the ones in the existing Google sheet schedule - including Keynote events, Custom events, Talk events, Tutorial events
2. Go to the `/events/schedule/new` to review the laid out schedule
3. Click on the **Generate** button at the bottom of the `/events/schedule/new`
4. See if the new schedule is published at the `/events/schedule`

## Related Issue
#813

## More Information
**Screenshots**

- Locate at `/events/schedule`
image

- Scroll down the schedule timetable
image

- Hover an event (also hover at the long-line title)
image

### CODE REVIEW

**Comment by yychen** [comment]
圖片
我把
```css
grid-template-rows: repeat(156, 1fr);
```
槓掉就秒變 compact 耶!!!! !!!

**Comment by yychen** [comment]
I think we can have a toggle option fixed on the page, like, at the upper-left corner. We let the user toggle on / off the normal & compact mode. Compact mode actually looks nice to me.

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+					.context-container {
+						position: sticky;
+						top: 70px;
+						padding: 12px;
+
+						.context {
+							.title {
+								margin-bottom: 8px;
+								font-size: 22px;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:108]: I think the font is too large

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+	.py-schedule {
+		&-tabs {
+			display: flex;
+			justify-content: center;
+			align-items: center;
+
+			&__tab {
+				margin: 0 24px 24px;
+				padding: 24px 72px;
+				font-family: Philosopher, sans-serif;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:27]: Do you consider to use something like `$header-font-family` instead?

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+				z-index: 1000;
+				display: flex;
+				background: transparentize($brick, 0.3);
+
+				&-column {
+					display: flex;
+					margin: 8px 8px 8px 0;
+					padding: 12px 0;
+					justify-content: center;
+					font-family: Quicksand, sans-serif;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:57]: This one with `$text-font-family`?

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+						[room-5-r1-end room-6-r2-start] 2fr
+					  [room-6-r2-end room-8-r3-start] 2fr
+					  [room-8-r3-end room-7-r4-start] 2fr
+					  [room-7-r4-end];
+				grid-column-gap: 8px;
+				grid-row-gap: 8px;
+
+				&-item {
+					display: flex;
+					font-family: Quicksand, sans-serif;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:97]: This one also

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+				}
+
+				.room-2-all {
+					grid-column: room-4-r0-start / room-8-r3-end;
+				}
+
+				.timeline {
+					grid-column: times;
+					justify-self: center;
+					font-family: Philosopher, sans-serif;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:280]: This one also

File: src/templates/pycontw-2020/_includes/menu.html [hunk 6]
```diff
@@ -6,7 +6,7 @@
 {% url 'page' path='about/community' as about_community_url %}
 {% url 'page' path='about/staff' as about_staff_url %}
 {% url 'page' path='conference/overview' as events_overview_url %}
-{% url 'events_schedule' as events_schedule_url %}
+{% url 'page' path='events/schedule' as events_schedule_url %}
```
Comment (yychen) [inline src/templates/pycontw-2020/_includes/menu.html:9]: Why change to this?

File: src/static/pycontw-2020/styles/pages/_events.scss [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+						}
+					}
+
+					.snake {
+						@include icon('snake', 0, center);
+
+						&::before {
+							background-size: contain;
+						}
+					}
```
Comment (yychen) [inline src/static/pycontw-2020/styles/pages/_events.scss:246]: Is it possible that we change it to this year's theme? like the window pattern or the ancient tower (赤崁樓) icon? The grass and snake is kind of the Pokemon's theme for last year.

**Comment by yychen** [comment]
And, I think you'll also need to change the `src/events/models.py`, hack the names into the following:
```py
    LOCATION_CHOICES = [
        (Location.ALL,  _('All rooms')),
        # (Location.R012, _('R0, R1, R2')),
        # (Location.R0,   _('R0')),
        # (Location.R1,   _('R1')),
        # (Location.R2,   _('R2')),
        # (Location.R3,   _('R3')),
        # (Location.R4,   _('R4')),

        (Location.R012, _('R1, R2, R3')),
        (Location.R0,   _('R1')),
        (Location.R1,   _('R2')),
        (Location.R2,   _('R3')),
        (Location.R3,   _('Multifunction room')),
        (Location.R4,   _('Goodideas Studio')),
    ]
```

**Comment by flynnhou** [comment]
@yychen based on your comments above, I've updated the UI and some refactors.
- In short, the UI would look like this with a new simple button to toggle the timetable's grid mode between `default` or `compact`.

image

- The rest of the updates please check out the commits I'd pushed 🙏. I'm too lazy to reply to each of your comment 😛

**Comment by flynnhou** [comment]
@yychen I've rebased the master branch thus the forced push.

Based on the offline discussions regarding the style of the schedule, I've modified the below changes:
(1) Remove the unnecessary icon decoration of the custom events - break, lunch, etc.
(2) Set the grid to compact mode by default for it's unnecessary to reflect the actual time length of an event
(3) Take out the text decoration on `<a>`
(4) Change the events with better contrast background colors

photo_2020-08-28_01-13-42

**Comment by flynnhou** [comment]
- Improved desktop schedule styles
image

- Added mobile view schedule
image
image

**Review by yychen — APPROVED** [review]
They look very nice! Thank you for your hard work.

**Pull request merged**